ui-admin/src/study/kits/KitList.tsx

```diff
@@ -1,33 +1,96 @@
+import Api, { KitRequest, PepperKitStatus } from 'api/api'
 import { StudyEnvContextT } from 'study/StudyEnvironmentRouter'
 import LoadingSpinner from 'util/LoadingSpinner'
-import { basicTableLayout, FilterType } from 'util/tableUtils'
+import { basicTableLayout, ColumnVisibilityControl } from 'util/tableUtils'
 import { instantToDateString, isoToInstant } from 'util/timeUtils'
 
-const pepperStatusToHumanStatus = (dsmStatus?: string): string => {
-  const statusMap: Record<string, string> = {
-    'CREATED': 'Created',
-    'LABELED': 'Prepared',
-    'SCANNED': 'Sent',
-    'RECEIVED': 'Returned'
+type KitStatusTabConfig = {
+  status: string,
+  label: string,
+  additionalColumns?: string[]
+}
+
+const defaultColumns = {
+  'enrollee_shortcode': true,
+  'kitType_displayName': true,
+  'createdAt': true,
+  'pepperStatus_labelDate': false,
+  'pepperStatus_trackingNumber': false,
+  'pepperStatus_scanDate': false,
+  'pepperStatus_returnTrackingNumber': false,
+  'pepperStatus_receiveDate': false,
+  'pepperStatus': false
+}
+
+const statusTabs: KitStatusTabConfig[] = [
+  {
+    status: 'CREATED',
+    label: 'Created',
+    additionalColumns: []
+  },
+  {
+    status: 'LABELED',
+    label: 'Labeled',
+    additionalColumns: [
+      'pepperStatus_labelDate', 'pepperStatus_trackingNumber'
+    ]
+  },
+  {
+    status: 'SCANNED',
+    label: 'Sent',
+    additionalColumns: [
+      'pepperStatus_labelDate', 'pepperStatus_trackingNumber',
+      'pepperStatus_scanDate', 'pepperStatus_returnTrackingNumber'
+    ]
+  },
+  {
+    status: 'RECEIVED',
+    label: 'Returned',
+    additionalColumns: [
+      'pepperStatus_labelDate', 'pepperStatus_trackingNumber',
+      'pepperStatus_scanDate', 'pepperStatus_returnTrackingNumber',
+      'pepperStatus_receiveDate'
+    ]
+  },
+  {
+    status: 'ERROR',
+    label: 'Issues',
+    additionalColumns: [
+      'pepperStatus_labelDate', 'pepperStatus_trackingNumber',
+      'pepperStatus_scanDate', 'pepperStatus_returnTrackingNumber',
+      'pepperStatus_receiveDate',
+      'pepperStatus'
+    ]
   }
-  return dsmStatus
-    ? (statusMap[dsmStatus] || (`(${  dsmStatus  })`))
-    : '(unknown)'
+]
+
+const initialColumnVisibility = (tab: KitStatusTabConfig): VisibilityState => {
+  return {
+    ...defaultColumns,
+    ..._fromPairs(tab.additionalColumns?.map(c => { return [c, true] }))
+  }
+}
+
+const pepperStatusToHumanStatus = (pepperStatus?: PepperKitStatus): string => {
+  const tab = statusTabs.find(tab => tab.status === pepperStatus?.currentStatus)
+  return tab?.label || pepperStatus?.currentStatus || '(unknown)'
 }
 
 /** Loads sample kits for a study and shows them as a list. */
 export default function KitList({ studyEnvContext }: { studyEnvContext: StudyEnvContextT }) {
   const { portal, study, currentEnv } = studyEnvContext
   const [isLoading, setIsLoading] = useState(true)
+  const [activeTab, setActiveTab] = useState<string | null>(statusTabs[0].status)
```

Comment:
this should be a url param so the tabs are linkable.  I could see this either being a path param or a query param